### DISCUSSION TITLE
Generated Migration When noImplicitOverride is Enabled Compile Fail Solution

### DISCUSSION THREAD

**Discussion by arinanto** [mikro-orm/mikro-orm#5634] [Q&A] [answered]
TypeScript failed to compile with error when `noImplicitOverride` is set to true with the following error:
```
src/providers/database/migrations/Migration20240530085839.ts:13:9 - error TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'Migration'.

13   async down(): Promise<void> {
           ~~~~

Found 1 error(s)
```
This is due to the generated migration doesn't have the `override` keyword in its `down()` method.

Is there any solution to this without disabling `noImplicitOverride`?

**Comment by B4nan** [comment]
You can provide your own migration generator that will do this.

https://mikro-orm.io/docs/migrations#using-custom-migrationgenerator

**Reply by B4nan** [reply]
Also, I think you could just add another tsconfig into your migrations folder and disable that option locally only for the migrations.

**Reply by simonbrunel** [reply]
Also getting a warning / error: @B4nan is there a reason to not explicitly add the `override` keyword in the built-in [TSMigrationGenerator](https://github.com/mikro-orm/mikro-orm/blob/master/packages/migrations/src/TSMigrationGenerator.ts#L16)? This feature has been introduced since TS 4.3, thus should be not breaking (MikroORM 6+ requires TypeScript 5+).

**Reply by B4nan** [reply, marked as answer]
Should be fine I guess, here you go d2f7b83557431da53c0fe0c2011cddee67ceac05

**Reply by simonbrunel** [reply]
Thanks @B4nan!
